acceptance/tests/terminating-gateway/terminating_gateway_destinations_test.go

```diff
@@ -0,0 +1,178 @@
+package terminatinggateway
+
+import (
+	"fmt"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-version"
+	"github.com/stretchr/testify/require"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+)
+
+// Test that egress Destinations route through terminating gateways in transparent mode.
+func TestTerminatingGatewayDestinations(t *testing.T) {
```

Comment:
Is there a way this test can be merged into the existing terminating gateway tests with a toggle for `transparentProxy` enabled and disabled, where it follows the test flow of this test when tproxy is enabled but the existing flow when disabled? that way we would have coverage for when consul enterprise namespaces are enabled as well.

That is the pattern we have used for tests that need to run in transparent and "non-transparent" proxy modes so far.

Reply:
This test is specific for destinations. Destinations are not supported other than in tproxy. I feel if we merge all the use cases in the terminating gateway tests it will make it hard to read with a lot of conditionals.

Reply:
Thanks for the explanation Dhia! Can you add that to the comment above the test so that we have that context in the future when working on the terminating gateways tests?